feathr_project/feathr/client.py

```diff
@@ -568,13 +569,58 @@ def monitor_features(self, settings: MonitoringSettings, execution_configuration
         """
         self.materialize_features(settings, execution_configurations, verbose)
 
+    # Get feature keys givin the name of a feature
+    # Should search in both 'derived_feature_list' and 'anchor_list'
+    # Return related keys(key_column list) or None if cannot find the feature
+    def _get_feature_key(self, feature_name: str):
+        features = []
+        if 'derived_feature_list' in dir(self):
+            features += self.derived_feature_list 
+        if 'anchor_list' in dir(self):
+            for anchor in self.anchor_list:
+                features += anchor.features  
+        for feature in features:
+            if feature.name == feature_name:
+                keys = feature.key
+                return [key.key_column for key in keys] 
+        self.logger.warning(f"Invalid feature name: {feature_name}. Please call FeathrClient.build_features() first in order to materialize the features.")
+        return None
+        
+    # Validation on feature keys:
+    # Features within a set of aggregation or planned to be merged should have same keys
+    # The param "allow_empty_key" shows if empty keys are acceptable 
+    def _valid_materialize_keys(self, features: List[str], allow_empty_key=False):
+        keys = None
+        for feature in features:
+            new_keys = self._get_feature_key(feature)
+            if new_keys is None:
+                self.logger.error(f"Failed to get feature key for feature: {feature}")
```

Comment:
Say something like this to give clear directions to users:

Key of feature: {feature} is empty. If this feature is not from INPUT_CONTEXT, you might want to double check on the feature definition to see whether the key is empty or not.

Reply:
Changed.